ibis/expr/types/generic.py

```diff
@@ -1271,9 +1272,69 @@ def __array__(self, dtype=None):
         return self.execute().__array__(dtype)
 
     def __interactive_rich_console__(self, console, options):
+        return console.render(self.preview(), options=options)
+
+    def preview(
+        self,
+        *,
+        max_rows: int | None = None,
+        show_types: bool | None = None,
+        max_length: int | None = None,
+        max_string: int | None = None,
+        max_depth: int | None = None,
+        console_width: int | float | None = None,
```

Comment:
So one of the motivators for this PR was better UX when using the VSCode debugger. Currently this PR doesn't quite work yet:
<img width="340" alt="image" src="https://github.com/ibis-project/ibis/assets/10820686/fa4353bd-7360-4690-a08c-f0a76a6fb39e">
Ideally this would show the table nicely. But I noticed when I did the workaround of converting to pandas, that the console width seems to be small and I get ugly wrapping:
<img width="625" alt="image" src="https://github.com/ibis-project/ibis/assets/10820686/6188c8e5-ac1c-4779-bd0f-96ea7480ae74">

I was guessing that this `console_width` param was needed so the output would be pretty. But perhaps I was getting ahead of myself. we leave it out for now, and cross that bridge when we come to it?